argilla/src/argilla/datasets/_export/_disk.py

```diff
@@ -90,28 +90,39 @@ def from_disk(
 
         # Get the relevant workspace_id of the incoming dataset
         if isinstance(workspace, str):
-            workspace_id = client.workspaces(workspace).id
-        elif isinstance(workspace, Workspace):
-            workspace_id = workspace.id
+            workspace = client.workspaces(workspace)
+            if not workspace:
+                raise ArgillaError(f"Workspace {workspace} not found on the server.")
         else:
             warnings.warn("Workspace not provided. Using default workspace.")
```

Comment:
Isn't this warning already introduced when creating a dataset in general?

Reply:
It is. But we use the `workspace_id` outside of the `Dataset` resource.